app/models/laboratory.rb

```diff
@@ -6,13 +6,22 @@ class Laboratory < ApplicationRecord
 
   validates :result, inclusion: { in: ['positive', 'negative', 'indeterminate', 'other', nil, ''] }
 
-  scope :last_ten_days_positive, lambda {
-    where('specimen_collection > ?', 10.days.ago).where(result: 'positive')
-  }
+  after_save :update_patient_linelist_after_save
+  before_destroy :update_patient_linelist_before_destroy
 
-  scope :before_ten_days_positive, lambda {
-    where('specimen_collection <= ?', 10.days.ago).where(result: 'positive')
-  }
+  def update_patient_linelist_after_save
+    patient.update(
+      latest_positive_lab_at: patient.laboratories.where(result: 'positive').maximum(:specimen_collection),
```

Comment:
same reasoning as above in that it needs to handle the case of updates. on top of that, the specimen collection date might not necessarily be in the same order as the time laboratories are created. there's also currently no validation when it comes to creating laboratories so it's also possible to not have a specimen collection date

Reply:
Well that's kind of annoying about no date...So it seems like this is recording the latest positive lab which also has a date provided? Is that what should be displayed? Did you give any consideration to latest positive lab period (use created at as fallback if collection date doesn't exist?)

Reply:
this was actually originally based off of the report date field (which is also just a date instead of datetime), but Kelly and the users specifically wanted it to be off of the specimen collection date. I'm not sure if they would want a fallback field to be used because that might end up being confusing if different fields are used to determine the status between different patients

Reply:
https://jira.mitre.org/browse/SARAALERT-514